src/readcalls.c

```diff
@@ -84,105 +80,54 @@ void init_scoring(void) {
 
     InitPfx();
 
-    if (pfxnummultinr > 0) {
-	for (int i = 0; i < pfxnummultinr; i++) {
-	    for (int n = 0; n < PFXNUMBERS; n++) {
-		pfxnummulti[i].qsos[n] = 0;
-	    }
+    for (int i = 0; i < pfxnummultinr; i++) {
+	for (int n = 0; n < PFXNUMBERS; n++) {
+	    pfxnummulti[i].qsos[n] = 0;
 	}
     }
 }
 
 static void show_progress(int linenr) {
+    if (linenr == 1) {
+	printw(spaces(2));  // leading separator after log file name
+	refreshp();
+    }
     if (((linenr + 1) % 100) == 0) {
 	printw("*");
 	refreshp();
     }
 }
 
-
-/* pick up multi string from logline
- *
- * ATTENTION! return value needs to be freed
- */
-char *get_multi_from_line(char *logline) {
-    char *multbuffer = g_malloc(20);
-
-    if (CONTEST_IS(ARRL_SS)) {
-
-	if (logline[63] == ' ')
-	    g_strlcpy(multbuffer, logline + 64, 4);
-	else
-	    g_strlcpy(multbuffer, logline + 63, 4);
-
-    } else if (serial_section_mult
-	       || sectn_mult
-	       || sectn_mult_once) {
-
-	g_strlcpy(multbuffer, logline + 68, MAX_SECTION_LENGTH + 1);
-	g_strchomp(multbuffer);
-
-    } else if (serial_grid4_mult) {
-
-	g_strlcpy(multbuffer, logline + 59, 5);
-
-    } else if (unique_call_multi != 0) {
-
-	g_strlcpy(multbuffer, logline + 68, 10);
-	g_strchomp(multbuffer);
-
-    } else {
-
-	g_strlcpy(multbuffer, logline + 54, 11);	// normal case
-	g_strchomp(multbuffer);
-
-    }
-
-    return multbuffer;
-}
-
-
-void count_if_worked(int check, unsigned int bandindex, int *count) {
-    if (bandindex >= NBANDS)
-	return;
-
-    if ((check & inxes[bandindex]) != 0)
-	count[bandindex]++;
-}
-
-void count_contest_bands(int check, int *count) {
-    count_if_worked(check, BANDINDEX_160, count);
-    count_if_worked(check, BANDINDEX_80, count);
-    count_if_worked(check, BANDINDEX_40, count);
-    count_if_worked(check, BANDINDEX_20, count);
-    count_if_worked(check, BANDINDEX_15, count);
-    count_if_worked(check, BANDINDEX_10, count);
+static void qso_free(gpointer data) {
+    free_qso((struct qso_t *) data);
 }
 
-
-
-int readcalls(const char *logfile) {
+int readcalls(const char *logfile, bool interactive) {
 
     char inputbuffer[LOGLINELEN + 1];
     struct qso_t *qso;
     int linenr = 0;
 
     FILE *fp;
 
-    char *info = g_strdup_printf("Reading logfile: %s\n", logfile);
-    showmsg(info);
-    g_free(info);
-    refreshp();
+    if (interactive) {
+	showstring("Reading logfile:", logfile);
+    }
 
     init_scoring();
 
     if ((fp = fopen(logfile, "r")) == NULL) {
 	showmsg("Error opening logfile ");
-	refreshp();
 	sleep(2);
 	exit(1);
     }
 
+    bool log_changed = false;
+    // array of qso's
+    // FIXME use this instead of qsos[] (make it global and don't free it here)
+    // FIXME also include comments/notes into qso_t
```

Comment:
We could just add a 'iscomment' flag and a pointer to the comment string or save some memory by using a union union from the actual qso_t and the comment field pointer.